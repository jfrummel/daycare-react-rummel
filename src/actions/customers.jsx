import { v4 as uuidv4 } from 'uuid';

//  Customers action generators

export const addCustomer = ({ firstName = "", lastName = "", numOfChildren = 0, hourlyRate = 0 } = {}) => ({
    type: "ADD_CUSTOMER",
    customer: {
        id: uuidv4(),
        firstName,
        lastName,
        numOfChildren,
        hourlyRate
    }
});

export const removeCustomer = ({ id } = {}) => ({
    type: "REMOVE_CUSTOMER",
    id
});

export const editCustomer = (id, updates) => ({
    type: "EDIT_CUSTOMER",
    id, updates
});