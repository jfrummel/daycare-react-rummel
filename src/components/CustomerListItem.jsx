import React from 'react';
import { Link } from 'react-router-dom';

const CustomerListItem = (props) => (
    <li>
        <Link to={`edit/${props.customer.id}`}>{props.customer.firstName} {props.customer.lastName}</Link>
        <p>Number of children: {props.customer.numOfChildren}</p>
        <p>Hourly rate: ${props.customer.hourlyRate}</p>
    </li>
);

export default CustomerListItem;