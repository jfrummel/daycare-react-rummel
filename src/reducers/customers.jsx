//  customers default state

const customerDefaultState = [];

const customerReducer = (state = customerDefaultState, action) => {
    switch (action.type) {
        case "ADD_CUSTOMER":
            return [...state, action.customer];
        case "REMOVE_CUSTOMER":
            return state.filter(({ id }) => id !== action.id);
        case "EDIT_CUSTOMER":
            return state.map((customer) => {
                if (customer.id === action.id) {
                    return { ...customer, ...action.updates };
                } else {
                    return customer
                }
            });
        default:
            return state;
    }
};

export default customerReducer;