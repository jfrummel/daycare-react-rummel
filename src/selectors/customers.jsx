
export const getVisibleCustomers = (customers, { text }) => {
    return customers.filter((customer) => {
        const textMatch = customer.lastName.toLowerCase().includes(text.toLowerCase()) ||
            customer.firstName.toLowerCase().includes(text.toLowerCase());

        return textMatch;
    });
};