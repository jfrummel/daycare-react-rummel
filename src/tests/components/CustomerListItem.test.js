import React from 'react';
import { shallow } from 'enzyme';
import CustomerListItem from '../../components/CustomerListItem';
import { customers } from '../fixtures/customers';

test("renders list item with correct data", () => {
    const wrapper = shallow(<CustomerListItem customer={customers[0]} />);
    expect(wrapper).toMatchSnapshot();
});
