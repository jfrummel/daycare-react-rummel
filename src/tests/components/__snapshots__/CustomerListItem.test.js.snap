// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list item with correct data 1`] = `
<li>
  <Link
    to="edit/1"
  >
    Trajan
     
    Alexander
  </Link>
  <p>
    Number of children: 
    3
  </p>
  <p>
    Hourly rate: $
    1
  </p>
</li>
`;
